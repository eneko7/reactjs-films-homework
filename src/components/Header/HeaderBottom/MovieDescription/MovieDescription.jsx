import React from 'react';
import PropTypes from 'prop-types';
import style from './MovieDescription.scss';

const MovieDescription = (props) => {
  const { filmDescrProps, filmName } = props;
  const filmDescr = filmDescrProps.split(',');
  const descrItems = filmDescr.map(item => (
    <span
      key={item}
      className={style.full_description_filmDescr_item}
    >
      {item}
    </span>
  ));
  return (
    <div className={style.full_description}>
      <span className={style.full_description_filmName}>{filmName}</span>
      <div className={style.full_description_filmDescr}>
        {descrItems}
      </div>
    </div>
  );
};

MovieDescription.propTypes = {
  filmName: PropTypes.string.isRequired,
  filmDescrProps: PropTypes.string.isRequired,
};

export default MovieDescription;
