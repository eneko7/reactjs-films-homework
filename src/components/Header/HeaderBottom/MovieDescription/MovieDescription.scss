.full_description {
  width:100%;
  
  .full_description_filmName {
    font-size: 45px;
    letter-spacing: 1px;
    line-height: 44px;
    color: #fff;
    font-family: Helvetica, Arial, Verdana, Tahoma, sans-serif;
    font-weight: 800;
    display: block;
  }

  .full_description_filmDescr {
    margin-top: 15px;
    display: flex;
    justify-content: space-between;

    .full_description_filmDescr_item {
      display: flex;
      font-size: 18px;
      line-height: 30px;
      color: #fff;
      font-family: Helvetica, Arial, Verdana, Tahoma, sans-serif;
      margin-right: 15px;
      font-weight: normal;
    }
  }
}

@media (max-width: 900px) {
  .full_description {
    .full_description_filmDescr {
      display: block;

      .full_description_filmDescr_item {
        display: inline-block;
        font-size: 15px;
        margin-right: 10px;
      }
    }
  }
}

@media (max-width: 767px) {
  .full_description {
    .full_description_filmName {
      font-size: 25px;
    }

    .full_description_filmDescr {
      display: block;
      margin-top: 5px;

      .full_description_filmDescr_item {
        display: inline-block;
        font-size: 15px;
        line-height: 20px;
        margin-right: 7px;
      }
    }
  }
}
