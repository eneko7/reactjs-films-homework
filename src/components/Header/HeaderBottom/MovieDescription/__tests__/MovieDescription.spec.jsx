import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import MovieDescription from '../index';

const data = {
  filmName: 'name',
  filmGenres: [
    {
      name: 'Action',
      id: 28,
    },
    {
      name: 'Animation',
      id: 16,
    },
    {
      name: 'Comedy',
      id: 35,
    },
  ],
};

describe('MovieDescription', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<MovieDescription {...data} />);
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
