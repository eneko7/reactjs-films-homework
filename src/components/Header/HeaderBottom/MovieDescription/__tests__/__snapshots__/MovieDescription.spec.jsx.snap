// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDescription renders correctly 1`] = `
<div
  className="full_description"
>
  <span
    className="full_description_filmName"
  >
    THE JUNGLE BOOK
  </span>
  <div
    className="full_description_filmDescr"
  >
    <span
      className="full_description_filmDescr_item"
    >
      Adventure
    </span>
    <span
      className="full_description_filmDescr_item"
    >
       Drama
    </span>
    <span
      className="full_description_filmDescr_item"
    >
       Family
    </span>
    <span
      className="full_description_filmDescr_item"
    >
       Fantasy
    </span>
    <span
      className="full_description_filmDescr_item"
    >
       |
    </span>
    <span
      className="full_description_filmDescr_item"
    >
       1h 46m
    </span>
  </div>
</div>
`;
